src/frontend/screens/AuthScreen.tsx

```diff
@@ -0,0 +1,197 @@
+  const [error, setError] = React.useState(false);
+  const {
+    setKillState,
+    killState,
+    setCheckFlag,
+    passcode,
+    killStateActive,
+  } = React.useContext(SecurityContext);
+  const ref = useBlurOnFulfill({ value: inputtedPass, cellCount: CELL_COUNT });
+
+  const [props, getCellOnLayoutHandler] = useClearByFocusCell({
+    value: inputtedPass,
+    setValue: setInputtedPass,
+  });
+  const userPass = React.useRef("");
+
+  React.useEffect(() => {
+    if (inputtedPass.length === CELL_COUNT) {
+      if (validatePassword(inputtedPass)) {
+        if (inputtedPass === KILL_PASSCODE && killStateActive) {
+          setKillState(true);
+          setCheckFlag(false);
+          return;
+        }
+
+        if (inputtedPass === passcode) {
+          if (killState) setKillState(false);
+          setCheckFlag(false);
+          return;
+        }
+
+        setError(true);
+      }
+
+      setInputtedPass("");
+    }
+  }, [inputtedPass]);
+
+  React.useEffect(() => {
+    AsyncStorage.getItem(PASSWORD_KEY, (err, result) => {
+      if (!!err) return;
+
+      userPass.current = result || "";
+    });
+  }, []);
+
+  function validateAndSetInput(text: string) {
+    if (onlyNumRegEx.test(text)) {
+      setInputtedPass(text);
+    }
+  }
+
+  const renderCell = ({ index, symbol, isFocused }: RenderCellOptions) => {
+    let textChild = null;
+
+    if (symbol) {
+      textChild = (
+        <MaskSymbol
+          maskSymbol="*"
+          isLastFilledCell={isLastFilledCell({ index, value: inputtedPass })}
+        >
+          {symbol}
+        </MaskSymbol>
+      );
+    } else if (isFocused) {
+      textChild = <Cursor />;
+    }
+
+    return (
+      <Text
+        key={index}
+        style={[styles.cell, isFocused && styles.focusCell]}
+        onLayout={getCellOnLayoutHandler(index)}
+      >
+        {textChild}
+      </Text>
+    );
+  };
+
+  return (
+    <View style={[styles.container]}>
+      <Image source={require("../images/icon_mapeo_pin.png")} />
+      <Text style={[styles.title]}>Mapeo</Text>
+
+      <Text style={[{ marginBottom: 20, fontSize: 16 }]}>
+        <FormattedMessage {...m.enterPass} />
+      </Text>
+
+      <CodeField
+        ref={ref}
+        {...props}
+        value={inputtedPass}
+        caretHidden={true}
+        onChangeText={validateAndSetInput}
+        cellCount={CELL_COUNT}
+        rootStyle={styles.codeFieldRoot}
+        keyboardType="number-pad"
+        textContentType="oneTimeCode"
+        renderCell={renderCell}
+      />
+
+      {inputtedPass.length === 0 && error && (
+        <Text style={[styles.wrongPass]}>
+          <FormattedMessage {...m.wrongPass} />
+        </Text>
+      )}
+    </View>
+  );
+};
+
+function validatePassword(password: string) {
+  if (password.length !== CELL_COUNT) return false;
+
+  const passAsArray = Array.from(password);
+  return passAsArray.reduce((acc, letter) => {
+    return onlyNumRegEx.test(letter);
+  }, false);
```

Comment:
nit: would use [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) here instead

Reply:
better yet, just regex the entire string as is i.e. `onlyNumRegEx.test(password)`